### PR TITLE
[primTorch] Elementwise binary ops I tests

### PR DESCRIPTION
**Pull request by Aidyn-A**
This PR does not fix any issue. However it does make `opInfo` be aware of the issues #77742 and  #78389.

cc @mruberry

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -412,14 +413,29 @@ def _ref_test_helper(self, ctx, device, dtype, op):
                     exact_layout=True,
                     exact_is_coalesced=True,
                 )
-            except AssertionError as e:
-                # Raises the error if the precise dtype comparison wouldn't be
-                # different
-                if dtype is precise_dtype:
-                    raise e
+            except AssertionError:
+                # see issue #78389:
```
Comment (mruberry) [inline test/test_ops.py:417]: I looked at the linked issue (and added the primTorch label) but I'm not sure I understand -- 

where is `a.copy_(b)` called? And when `a` and `b` are the same dtype then `a != b` after an inplace copy?

Comment (Aidyn-A) [reply test/test_ops.py:417]: The `a` and `b` arguments are getting promoted in `elementwise_type_promotion_wrapper`. This promotion wrapper calls `_maybe_convert_to_dtype` that calls `prims.convert_element_type` which does the `copy`. I will prepare a code snippet to replicate the problem in the issue #78389. Apparently this is not a primTorch issue, but general pytorch

**Comment by mruberry** [comment]
> This is what I propose to go around the failing `tests_ops` due to vectorization (#77742 #78747 #78389 #79009). We can use `custom_tiny` for the functions that fail the tests at extreme values ( `div`, `fmod`, `remainder` and others). However, I strongly believe, we should fix the vectorization issue on fundamental level.
> 
> cc @IvanYashchuk @ngimel @mruberry

I'm still not sure I understand the exact issue -- is there a script that shows op-by-op exactly what's going on that's incorrect in PyTorch and causing the discrepancy with the refs?

**Comment by IvanYashchuk** [comment]
> > This is what I propose to go around the failing `tests_ops` due to vectorization (#77742 #78747 #78389 #79009). We can use `custom_tiny` for the functions that fail the tests at extreme values ( `div`, `fmod`, `remainder` and others). However, I strongly believe, we should fix the vectorization issue on fundamental level.
> > cc @IvanYashchuk @ngimel @mruberry
> 
> I'm still not sure I understand the exact issue -- is there a script that shows op-by-op exactly what's going on that's incorrect in PyTorch and causing the discrepancy with the refs?

In https://github.com/pytorch/pytorch/issues/79009 there's an example script to reproduce the problem with `torch.fmod`.
bfloat16 support for fmod on CPU was added in https://github.com/pytorch/pytorch/pull/61897 with immediate skips of a few OpInfo tests.
As far as I understand a lot of vectorized kernels use the SLEEF library which seems to be the root cause.

**Comment by Aidyn-A** [comment]
> > This is what I propose to go around the failing `tests_ops` due to vectorization (#77742 #78747 #78389 #79009). We can use `custom_tiny` for the functions that fail the tests at extreme values ( `div`, `fmod`, `remainder` and others). However, I strongly believe, we should fix the vectorization issue on fundamental level.
> > cc @IvanYashchuk @ngimel @mruberry
> 
> I'm still not sure I understand the exact issue -- is there a script that shows op-by-op exactly what's going on that's incorrect in PyTorch and causing the discrepancy with the refs?

Hope this script will be more informative:

```
import torch
import torch._refs as refs

def fmod_with_copy(a, b):
    # This is actually happening in primTorch
    a_clone = torch.empty_like(a, dtype = torch.float32)
    b_clone = torch.empty_like(a, dtype = torch.float32)
    a_clone.copy_(a)
    b_clone.copy_(b)

    return torch.fmod(a_clone, b_clone).to(torch.bfloat16)


def check_fmod(n, dtype = torch.float32, device = 'cpu'): 
    a = 10*torch.randn(n, device = device, dtype = dtype)
    b = torch.randn(n, device = device, dtype = dtype)

    random_index = torch.randint(0, n, size = (1,)).item()
    a[random_index] = torch.tensor(10.0, device = device, dtype = dtype)
    b[random_index] = torch.tensor(2.0e-38, device = device, dtype = dtype) 

    res1 = torch.fmod(a, b)
    res2 = refs.fmod(a, b)
    res3 = fmod_with_copy(a, b)

    nans1 = res1.isnan().sum()
    nans2 = res2.isnan().sum()
    nans3 = res3.isnan().sum()

    if not nans1 == nans2 or not nans1 == nans3:
        print(res1[random_index])
        print(res2[random_index])
        print(res3[random_index])
        print()

for n in range(1, 64):
    check_fmod(n, dtype = torch.bfloat16, device = 'cpu')
```

The expected output for all three operations should be exactly the same. 
Please run the script on Linux machine, as this issue was not observed on Windows.

**Comment by ngimel** [comment]
We should stop reference testing half on cpu, and we should liberally skip bfloat16 reference tests on cpu, unless they just work.

**Comment by Aidyn-A** [comment]
> We should stop reference testing half on cpu, and we should liberally skip bfloat16 reference tests on cpu, unless they just work.

Closing the PR.
